### ISSUE TITLE
npm install seems to ignore global .gitconfig when using implicit Github URLs

### ISSUE THREAD

**Issue by callumlocke** [npm/npm#8131]
We're working on an app that uses [implicit Github URLs](https://docs.npmjs.com/files/package.json#github-urls) in its dependencies, like this:

```json
{
  "dependencies": {
    "swig": "kavanagh/swig"
  }
}
```

And we're behind a corporate firewall, so all of us have a global `.gitconfig` containing this:

```
[url "https://"]
	insteadOf = git://
```

This works fine for everyone except one guy. On his machine, `npm install` can't fetch the kavanagh/swig repo. Here is his [npm-debug.log](http://pastie.org/private/4mffusr5zefa1bfomh83zw#140-141,143,146-147,149-150,154,230-231,295-297,301-314).

The strange thing is when I change all these Github dependencies `git://` URLs, like `"swig": "git://github.com/kavanagh/swig"`, npm install works fine.

It seems that using the implicit-style `username/repo` URL causes npm to bypass his global `.gitconfig` – it seems to try for a couple of minutes then give up, as if it can't get through the firewall. (Then again in the debug log it looks like it is using the HTTPS protocol, so maybe it's something else.)

This only happens for one team member and he's on the same platform as the rest of us (OS X Yosemite). I tried on his machine with both Node 0.11 and 0.12, same results.

Any ideas?

**Comment by smikes** [comment]
Thanks for the detailed bug report.  I imagine you've tried this, but can his account clone a https:// repository, like this -- ?

```
$ git clone git://github.com/kavanagh/swig
```

It looks like `npm` is getting an error `406 Not Acceptable` from github, which is a weird error - usually it means that the client is trying to send a format/encoding that the server doesn't accept (or vice versa).

Finally, can you try updating `npm` to the very latest version?  In the debug log, you're running with quite an old `npm`, 2.1.6 on node 0.11.  The stock npm with node 0.12.2 is 2.7.4, which is a few weeks behind the latest, 2.8.4.  Normally that wouldn't make a big difference but several bugs related to git repositories were fixed in the last few releases.  You can update `npm` with

```
sudo npm -g install npm@latest
```
(you may be able to omit sudo, depending on how your machine is set up.)

**Comment by callumlocke** [comment]
Yes `git clone git://github.com/kavanagh/swig` works fine on his machine. (If you remove the global `.gitconfig` converting git:// to https://, it stops working, as expected due to the firewall.)

I just tried switching to v0.12.2 as you suggest, and it works! I was surprised because v0.12.0 didn't work yesterday.

Just now I tried three different Node versions again on the machine in question, with `username/repo` URLs in the package.json...

- v0.11.15 - doesn't work
- v0.12.0 - doesn't work
- v0.12.2 - works

But, we've all been using v0.11.15 (because we're working on an app that needs this version) and only one out of the four of us have this problem. And we're all on the same OS (Yosemite). Weird bug.

We will have to keep using full github URLs until we can upgrade our app to v0.12, but this is not a big deal. Feel free to close this.
